service/worker/domainReplicationTaskHandler.go

```diff
@@ -65,8 +67,45 @@ func NewDomainReplicator(metadataManager persistence.MetadataManager, logger bar
 	}
 }
 
-// handleDomainReplicationTask handle the domain replication task
-func (domainReplicator *domainReplicatorImpl) HandleReceiveTask(task *replicator.DomainTaskAttributes) error {
+// HandleTransmissionTask handle transmission of the domain replication task
+func (domainReplicator *domainReplicatorImpl) HandleTransmissionTask(kafka messaging.Producer, domainOperation replicator.DomainOperation,
```

Comment:
Let's move this logic into a separate component which lives within frontend.

Reply:
done